tests/native_api_tests/test_api_model.cpp

```diff
@@ -39,6 +39,17 @@ void check_exception(PowerGridError const& e, PGM_ErrorCode const& reference_err
     std::string const err_msg{e.what()};
     CHECK(err_msg.find(reference_err_msg) != std::string::npos);
 }
+
+template <typename Func, class... Args>
+void error_handler(Func func, PGM_ErrorCode const& reference_error, std::string_view reference_err_msg,
+                   Args&&... args) {
+    try {
+        func(std::forward<Args>(args)...);
```

Comment:
```suggestion
        std::forward<Func>(func)(std::forward<Args>(args)...);
```
(May need explicit call to operator())
```suggestion
        std::forward<Func>(func).operator()(std::forward<Args>(args)...);
```